File: Stage16/Python/Problem18258.py
import sys
from collections import deque
input = sys.stdin.readline

def main():
    N = int(input())
    q = deque()
    
    for _ in range(N):
        command = input().split()
        
        if command[0] == "push":
            q.append(command[1])
        elif command[0] == "pop":
            print(q.popleft() if q else -1)
        elif command[0] == "size":
            print(len(q))
        elif command[0] == "empty":
            print(0 if q else 1)
        elif command[0] == "front":
            print(q[0] if q else -1)
        elif command[0] == "back":
            print(q[-1] if q else -1)

if __name__ == "__main__":
    main()

File: Stage16/Java/Problem18258.java
package Stage16.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;

public class Problem18258 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());

        Queue<Integer> queue = new LinkedList<>();
        int last = -1;

        for (int i = 0; i < N; i++) {
            String[] cmd = br.readLine().split(" ");

            switch (cmd[0]) {
                case "push":
                    last = Integer.parseInt(cmd[1]);
                    queue.offer(last);
                    break;
                case "pop":
                    sb.append(queue.isEmpty() ? -1 : queue.poll()).append("\n");
                    break;
                case "size":
                    sb.append(queue.size()).append("\n");
                    break;
                case "empty":
                    sb.append(queue.isEmpty() ? 1 : 0).append("\n");
                    break;
                case "front":
                    sb.append(queue.isEmpty() ? -1 : queue.peek()).append("\n");
                    break;
                case "back":
                    sb.append(queue.isEmpty() ? -1 : last).append("\n");
                    break;
            }
        }

        System.out.println(sb);
        br.close();
    }
}
